packages/neptune-extension/src/common/components/App.tsx

```diff
@@ -1,114 +1,76 @@
-    const platformNotebookContent = await platformNotebook.getContent();
-
-    const checkpoint = await leaderboardClient.api.createEmptyCheckpoint({
-      notebookId: notebook.id,
-      checkpoint: {
-        path: metadata.path,
-        // name, todo handle optional checkpoint name
-        // description, // todo handle optional checkpoint description
-      },
-    });
-
-    await leaderboardClient.api.uploadCheckpointContent({
-      id: checkpoint.id,
-      content: platformNotebookContent,
-    });
-
-    await platformNotebook.saveNotebookId(notebook.id);
-    setMetadata(platformNotebook.getMetadata());
-  };
+  const handleConfigure = () => setConfigureModalOpen(true);
 
   const handleUpload = async () => {
-    if (!metadata.notebookId) {
-      return;
-    }
-
-    const notebookContent = await platformNotebook.getContent();
-    const checkpoint = await leaderboardClient.api.createEmptyCheckpoint({
-      notebookId: metadata.notebookId,
-      checkpoint: {
-        path: metadata.path,
-        // name, // todo handle optional checkpoint name
-        // description, // todo handle optional checkpoint description
-      },
-    });
-
-    leaderboardClient.api.uploadCheckpointContent({
-      id: checkpoint.id,
-      content: notebookContent,
-    });
+    // eslint-disable-next-line no-console
+    console.debug('@implement me');
   };
 
-  const initialized = !!window.localStorage.getItem('neptune_api_token') && !!metadata.notebookId;
-
   return (
-    <Provider store={store}>
-      <div id="neptune-app">
-        <ToolbarWrapper>
-          <ToolbarButton
-            label="Configure"
-            title="Connect to Neptune"
-            icon="neptune"
-            compact={initialized}
-            onClick={handleConfigure}
-          />
-          <ToolbarButton
-            label="Upload"
-            title="Upload to Neptune"
-            icon="fa-cloud-upload"
-            visible={initialized}
-            onClick={handleUpload}
-          />
-        </ToolbarWrapper>
-      <Modal
-        isOpen={open}
-        onClose={() => setOpen(false)}
-      >
-        Welcome to Neptune!
-        <Input value="for test only" error />
-      </Modal>
-      </div>
-    </Provider>
+    <div id="neptune-app">
+      <ToolbarWrapper>
+        <ToolbarButton
+          label="Configure"
+          title="Connect to Neptune"
+          icon="neptune"
+          compact={isApiTokenValid}
+          onClick={handleConfigure}
+        />
+        <ToolbarButton
+          label="Upload"
+          title="Upload to Neptune"
+          icon="fa-cloud-upload"
+          visible={isApiTokenValid}
+          onClick={handleUpload}
+        />
+      </ToolbarWrapper>
+      {configureModalOpen && (
+        <ConfigureModal
+          onClose={() => setConfigureModalOpen(false)}
+        />
+      )}
+    </div>
   );
 };
 
 export default App;
 
+function validateGlobalApiToken() {
```

Comment:
I suggest moving this to sth like `/common/hooks/auth.ts`

Reply:
ok